### ISSUE TITLE
Multiparty doesn't work with Sails

### ISSUE THREAD

**Issue by super-iterator** [balderdashy/sails#3384]
I tried to use Skipper, but it buffers the uploaded files on disk, so I try now to use Multiparty, which works great with Express and bare Node.js

Multiparty works great to stream upload files without getting stored, yet, it fails with Sails, look at this:

```

module.exports = {

    image_uploadx : function  (req,res) {

        var form = new multiparty.Form();

        form.on('field', function(name, value) {
            console.log('field on ... ');
          if (name === 'path') {
            destPath = value;
          }
        });

        form.on('part', function(part) {
            console.log('on part ...');
            var date_string_seed = String(new Date())
            var random_name = require('crypto').createHash('sha1').update(date_string_seed).digest('hex');

            s3Client.putObject({
                Bucket: bucket,
                Key: random_name,
                ACL: 'public-read',
                Body: part,
                ContentLength: part.byteCount,
            }, function(err, data) {
                if (err){
                    throw err;
                } else {
                    console.log("done", data);
                    console.log("https://s3.amazonaws.com/" + bucket + '/' + random_name);
                }
            });
        });

        form.on('error',function  (err) {
            // res.send('error happpend: ' , err)
        });
    }
}
```

What shall be done now?

**Comment by sgress454** [comment]
Hi @super-iterator.  I can't really speak to your problems with Multiparty since 1) we're not developers on that project and 2) you haven't posted the actual problem you're having (you just said "it doesn't work").  Additionally, this forum is for reporting bugs on the Sails core, so this kind of question is more appropriate for [StackOverflow](http://stackoverflow.com), or our [Google Group](https://groups.google.com/forum/#!forum/sailsjs) or [online chat](https://gitter.im/balderdashy/sails?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge) (but again, you'll have to be a lot more explicit about the actual problems you're having).    

What I can say is that Skipper is specifically designed to _not_ buffer files to disk--that's the whole point.  The [skipper-s3 adapter](https://github.com/balderdashy/skipper-s3) does buffer _parts_ to disk, because Amazon requires parts to be at least 5 megabytes in size.  But otherwise Skipper uses streams and back-pressure to ensure that your disk (or memory) isn't overrun when uploading large files.

**Comment by super-iterator** [comment]
@sgress454 .. Thanks for your response!

Skipper-s3 adapter uses node-knox behind the scenes to stream-upload the files to S3, which never happens, I monitored the node process while uploading, it buffers the upload to disk.

While I tried the same for Node-Multiparty, example code in the link below: 
https://gist.github.com/super-iterator/de13aa572d2ca8a8aef2 

I monitored the process, no matter what how big the files are, nothing get buffered to disks, which is great.

Is that a difference between knox and multiparty regarding s3 stream-upload?
